File: sdk/agid-java/src/main/java/org/agid/Agid.java
package org.agid;

public final class Agid {
  public static final String BASE32_ALPHABET = "0123456789ABCDEFGHJKMNPQRSTVWXYZ";
  public static final int PREFIX_LENGTH = 2;
  public static final int HASH_LENGTH = 10;
  public static final int TOTAL_LENGTH = 12;

  private Agid() {}

  public static AgidResult encode(double lat, double lon) {
    throw new UnsupportedOperationException("wire this package to the AGID reference implementation");
  }

  public static AgidResult decode(String id) {
    return null;
  }

  public static AgidBounds cellBounds(String id) {
    throw new UnsupportedOperationException("wire this package to the AGID reference implementation");
  }
}

File: sdk/agid-nim/src/agid.nim
const
  base32Alphabet* = "0123456789ABCDEFGHJKMNPQRSTVWXYZ"
  prefixLength* = 2
  hashLength* = 10
  totalLength* = 12

type
  AgidResult* = object
    id*: string
    lat*: float
    lon*: float
    face*: int

  AgidBounds* = object
    minLat*: float
    maxLat*: float
    minLon*: float
    maxLon*: float

proc encode*(lat: float, lon: float): AgidResult =
  raise newException(CatchableError, "wire this package to the AGID reference implementation")

proc decode*(id: string): AgidResult =
  raise newException(CatchableError, "not implemented")

proc cellBounds*(id: string): AgidBounds =
  raise newException(CatchableError, "wire this package to the AGID reference implementation")
